src/webrtc/call.ts

```diff
@@ -411,10 +411,13 @@ export class MatrixCall extends TypedEventEmitter<CallEvent, CallEventHandlerMap
         this.invitee = opts.invitee;
         this.client = opts.client;
 
-        if (!this.client.deviceId) throw new Error("Client must have a device ID to start calls");
+        if (!this.client.getDeviceId()) throw new Error("Client must have a device ID to start calls");
 
         this.forceTURN = opts.forceTURN ?? false;
-        this.ourPartyId = this.client.deviceId;
+        this.ourPartyId = this.client.deviceId!;
+        if (!this.ourPartyId) {
+            logger.warn('MatrixCall: ourPartyId is null which may have unintended consequences');
+        }
```

Comment:
Random change but the strict tsc was complaining here.

There is probably a better value to fallback to here? `this.client.deviceId`  can be `null`

Reply:
Good point about the check above. I've updated it to use the same `const deviceId = this.client.getDeviceId();` reference so that the type inference can properly happen from that logic escape hatch

https://github.com/matrix-org/matrix-js-sdk/blob/4c49e68a865fc765e9028a1a2df9dcd7a1dd93ad/src/webrtc/call.ts#L414-L418